src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -31,49 +31,27 @@ public static class ActivatorUtilities
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type instanceType,
             params object[] parameters)
         {
-            int bestLength = -1;
-            bool seenPreferred = false;
-
-            ConstructorMatcher bestMatcher = default;
-
-            if (!instanceType.IsAbstract)
+            Type[] argumentTypes;
+            if (parameters == null || parameters.Length == 0)
+            {
+                argumentTypes = Type.EmptyTypes;
+            }
+            else
             {
-                foreach (ConstructorInfo? constructor in instanceType.GetConstructors())
+                argumentTypes = new Type[parameters.Length];
+                for (int index = 0; index < argumentTypes.Length; index++)
```

Comment:
```suggestion
                for (int index = 0; index < parameters.Length; index++)
```

Fix CQ issue:

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgIRgDNoYGBvGhgU0bFySBhGAArGGAwMAsgAoADtijZ8ucVJkYA2gF0Gq9fhgYYUXAEp+gvtUFOGAFQCeymIYZqA5gFczADsMd09cAG47ZwYAS3YGFTUNc0stAF50hiD/ABtchkJCY2SzCysAOgAZGCDfDAALBkyGAAZbRxjeaK6BP0Da0I8YDNdhioBRfGUMNzCRqM6YgF8epxhc3Bg1+x2Y/uCh8ObsmAB3Mc89ExTy3Gra+oaDRd6nTihE2JC4oIATGAIE6tCK/AFAgA8PigAUO83uNTqjVB33BAGo0R03rsltjBAdBvC9KjAUYsjcymlif9SRUAOLmeaKayvPGrXFOdl7JgAdmhsMJw0iPXZ7PoTFIDAAguxyt1ceKRGIJNJZAokqYtCrdN4KakrFicb15t4CSF4ayYvFEnq7s0sjl8oVira0g8kU0Wu1uQ48fyBuahSd5pNprMLdz2b0NlsfdynGajiMTkFzpcvK7Koini944IPl8fiSgVkQWDAQwoZmEY9keWEBjDdjfX78TCA0ncNTwWSSqZ9V3iwZ6YzhszLW8o10uRzBMQ+YmI7jRUA